src/components/auth/withAuthentication.js

```diff
@@ -1,35 +1,47 @@
 import React from 'react';
+import {connect} from 'react-redux'
+import { firebase, dbUsers } from '../../firebase';
+import {Dimmer, Segment} from 'semantic-ui-react'
 
-import AuthUserContext from './AuthUserContext';
-import { firebase } from '../../firebase';
-
-const withAuthentication = (Component) =>
-  class WithAuthentication extends React.Component {
+class WithAuthentication extends React.Component {
     constructor(props) {
       super(props);
 
       this.state = {
-        authUser: null,
+        loading: null,
```

Comment:
try set loading = true as initial run it should prevent to useless unauthorized app rendering